### ISSUE TITLE
Rolling back Liquidsoap version 1.4

### ISSUE THREAD

**Issue by gammaw** [AzuraCast/AzuraCast#2180]
Hi, we have an [issue](https://github.com/savonet/liquidsoap/issues/1009) with Liquidsoap 1.4 (and a related one [here](https://github.com/AzuraCast/AzuraCast/issues/1254)) but we would like to update to Azuracast 0.9.8 because of various issues (bad view performance of 0.9.7.1 and other). Is there a way to use Azuracast's latest version with previous versions of Liquidsoap?

**Comment by gammaw** [comment]
May be related, an interesting insight from the Liquidsoap developers: https://github.com/savonet/liquidsoap/issues/1009#issuecomment-559598553

**Comment by SlvrEagle23** [comment]
@gammaw We have built a special Docker image (and also tagged a special AzuraCast release as of the last commit before the 1.4.0 switch), so if it's absolutely necessary for you to be on version 1.3.7, you can put something like this in your `docker-compose.override.yml`:

```yaml
version: '2.2'
services:
  stations:
    image: azuracast/azuracast_stations:ls1.3.7
```

...and then run:

```bash
./docker.sh update dev-pre-1.4.0
```

Which should roll both your Stations Docker image and codebase back.

You won't get subsequent code improvements...right now we don't plan to support both Liquidsoap 1.3.7 and 1.4.0 in the new versions of the application.

**Comment by gammaw** [comment]
Wow thank you! We'll try to wait with updating our public server to 0.9.8 until the upstream Liquidsoap issues are fixed. They seem to be actively working on a solution.

**Comment by gammaw** [comment]
@SlvrEagle23 a related question is if there is way to use a newer version of Liquidsoap if it's not integrated into Azuracast yet? The reason I'm asking is because it seems that they (Liquidsoap) are planning a bug fix release and if it's possible to start experimenting with it before you officially upgrade to the new version?

**Comment by SlvrEagle23** [comment]
@gammaw It's a little tricky, but you could bind-mount in a replacement liquidsoap build into the location in the docker `stations` container where the Liquidsoap binary is located (which is currently `/var/azuracast/.opam/4.08.0/bin/liquidsoap`).

A bit of a pain, though. If there's a bug fix for issues from the 1.4 launch and it's a stable release, let me know and I'll work on getting it integrated ASAP.

**Comment by gammaw** [comment]
@SlvrEagle23 thanks, good to know! Thanks to the Liquidsoap folks and their help with a specific issue we could now finally update to the latest Azuracast version (with LS 1.4 in it). You may also want to checkout this fix (https://github.com/savonet/liquidsoap/issues/1009#issuecomment-566834819) in case you also have issues with fading. We use the `fade.final` and `fade.initial` scripts in our advanced LS configs inside Azuracast.
